pkg/hiveregistrator/util.go

```diff
@@ -0,0 +1,24 @@
+package hiveregistrator
+
+import (
+	"fmt"
+	"strings"
+)
+
+func ClusterNamespaceName(w WorkloadCluster) string {
+	// create a namespace name for a given cluster that is unique within a hive cluster (or set if multiple)
+	// region should not be necessary, but who knows
+	// TODO: any length constraints ? 63 chars ? 253 characterss?
+	// https://kubernetes.io/docs/concepts/overview/working-with-objects/names/
+
+	return strings.ToLower(fmt.Sprintf("aro-%s-%s-%s", w.Subscription, w.Location, w.Name))
+
+}
+
+func ClusterDeploymentName(w WorkloadCluster) string {
+	// create a cluster deployment name for a given cluster
+	// TODO: any length constraints ?
+
+	return strings.ToLower(fmt.Sprintf("cd-%s-%s", w.Subscription, w.Name))
```

Comment:
If we are going to have one cluster per namespace - we can probably just use cluster name? Or even hardcode something.

Reply:
Agreed, making both namespace and clusterdeployment name unique is not required